operationsgateway_api/src/routes/filters.py

```diff
@@ -130,16 +128,23 @@ async def update_favourite_filter(
     parameter
     """
 
+    if not name and not filter:
+        return "No updates necessary"
+
     token_payload = JwtHandler.get_payload(access_token)
     username = token_payload["username"]
 
     log.info("Updating favourite filter '%s' for user '%s", _id, username)
     log.debug("Name: %s, Filter: %s", name, filter)
 
+    filter_result = await FavouriteFilter.get_by_id(username, _id)
+    updated_name = name if name is not None or name == "" else filter_result.name
```

Comment:
Alternatively, maybe this should be checking truthyness like on line 131. Especially since trying to set name to `""` will fail once it hits:
https://github.com/ral-facilities/operationsgateway-api/blob/b653096244e07c8666240c67a35bdf5eef84e8f4/operationsgateway_api/src/users/favourite_filter.py#L44-L45

Currently, it is possible to set the filter to an empty string by:
```bash
curl -X 'PATCH'   'http://127.0.0.1:8000/users/filters/6655fb00434b426cbaed6370?name=n&filter='   -H 'accept: application/json'   -H 'Authorization: Bearer ...'
```
Since `name` being set allows us to pass the check on 131, and `"" is not None`. But it is not possible by:
```bash
curl -X 'PATCH'   'http://127.0.0.1:8000/users/filters/6655fb00434b426cbaed6370?&filter='   -H 'accept: application/json'   -H 'Authorization: Bearer ...'
```
Since `name is None` and `filter` evaluates to False. Not sure which is the intentional behaviour here.

Reply:
I've pushed d40acb4b553742c87a8207a2a001adfd94fd4bd7 so filters cannot be set to an empty string. Good spot about `name == ""`, I think I thought made a difference but I must've saved the file at the wrong time as it's obvious now that it wouldn't get evaluated.